src/guide/advanced/permissions.md

```diff
@@ -0,0 +1,286 @@
+value in the specified asset.
+
+### `CanRemoveKeyValueInUserAssets`
+
+With `CanRemoveKeyValueInUserAssets` permission token, a user can remove
+key value in the specified asset.
+
+### `CanSetKeyValueInUserMetadata`
+
+With `CanSetKeyValueInUserMetadata` permission token, a user can set key
+value in the metadata for the specified account.
+
+```rust
+let mouse_id = <Account as Identifiable>::Id::from_str("mouse@wonderland")?;
+
+// Create a new `CanSetKeyValueInUserMetadata` token that, when granted to another account,
+// allows it to set key value to the metadata in Mouse's account
+let permission_to_set_key_value_in_mouse_metadata: PermissionToken =
+    key_value::CanSetKeyValueInUserMetadata::new(mouse_id.clone()).into();
+```
+
+### `CanRemoveKeyValueInUserMetadata`
+
+With `CanRemoveKeyValueInUserMetadata` permission token, a user can remove
+key value in the metadata for the specified account.
+
+```rust
+let mouse_id = <Account as Identifiable>::Id::from_str("mouse@wonderland")?;
+
+// Create a new `CanRemoveKeyValueInUserMetadata` token that, when granted to another account,
+// allows it to remove key value from the metadata in Mouse's account
+let permission_to_remove_key_value_in_mouse_metadata: PermissionToken =
+    key_value::CanRemoveKeyValueInUserMetadata::new(mouse_id.clone()).into();
+```
+
+### `CanSetKeyValueInAssetDefinition`
+
+With `CanSetKeyValueInAssetDefinition` permission token, a user can set key
+value in the corresponding asset definition.
+
+### `CanRemoveKeyValueInAssetDefinition`
+
+With `CanRemoveKeyValueInAssetDefinition` permission token, a user can
+remove key value in the corresponding asset definition.
+
+### `CanRegisterDomains`
+
+with `CanRegisterDomains` permission token, a user can register domains.
+
+```rust
+let alice_id = AccountId::from_str("alice@test0")
+let mut alice = Account::new(alice_id.clone(), []).build();
+
+// Give Alice permission to register new domains
+alice.add_permission(register::CanRegisterDomains::new().into());
+```
+
+## Permission Groups (Roles)
+
+A set of permissions is called a **role**. Roles can be granted and
+revoked.
+
+<!-- TODO: add more info about roles -->
+
+### Register a new role
+
+Let's register a new role that, when granted, will allow another account
+access to the metadata in Mouse's account:
+
+```rust
+let role_id = <Role as Identifiable>::Id::from_str("ACCESS_TO_MOUSE_METADATA")?;
+let role = iroha_data_model::role::Role::new(role_id.clone())
+    .add_permission(CanSetKeyValueInUserMetadata::new(mouse_id.clone()))
+    .add_permission(CanRemoveKeyValueInUserMetadata::new(mouse_id.clone()));
+let register_role = RegisterBox::new(role);
+```
+
+### Grant a role
+
+After the role is registered, Mouse can grant it to Alice:
+
+```rust
+let grant_role = GrantBox::new(role_id.clone(), alice_id.clone());
+let grant_role_tx =
+    Transaction::new(mouse_id.clone(), vec![grant_role.into()].into(), 100_000)
+    .sign(mouse_key_pair)?;
```

Comment:
```suggestion
let grant_role = GrantBox::new(role_id, alice_id);
let grant_role_tx =
    Transaction::new(mouse_id, vec![grant_role.into()].into(), 100_000)
    .sign(mouse_key_pair)?;
```

Reply:
Considering this example is the extension of the example above and there aren't any more extensions let's remove extra-clonning here